### PR TITLE
3 regression tests available device #3335

### PR DESCRIPTION
**Pull request by BanzaiTokyo**
test_mean_absolute_relative_error.py
test_mean_error.py
test_mean_normalized_bias.py
test_median_absolute_error.py

### CODE REVIEW

File: tests/ignite/metrics/regression/test_median_absolute_error.py [hunk 1]
```diff
@@ -74,34 +72,33 @@ def test_median_absolute_error_2():
         idx = i * batch_size
         m.update((y_pred[idx : idx + batch_size], y[idx : idx + batch_size]))
 
-    assert np_median_absolute_error == pytest.approx(m.compute())
+    assert expected_median_absolute_error == pytest.approx(m.compute())
 
 
-def test_integration_median_absolute_error():
+def test_integration_median_absolute_error(available_device):
```
Comment (vfdev-5) [inline tests/ignite/metrics/regression/test_median_absolute_error.py:78]: Remove the line below `np.random.seed(1)`

File: tests/ignite/metrics/regression/test_mean_error.py [hunk 2]
```diff
@@ -26,76 +25,64 @@ def test_wrong_input_shapes():
...
+
+    total_error += torch.sum(ground_truth - c)
+    total_count += c.numel()
+    m.update((c, ground_truth))
+    assert m.compute() == pytest.approx((total_error / total_count).item())
+
+    total_error += torch.sum(ground_truth - d)
+    total_count += d.numel()
+    m.update((d, ground_truth))
+    assert m.compute() == pytest.approx((total_error / total_count).item())
```
Comment (vfdev-5) [inline tests/ignite/metrics/regression/test_mean_error.py:56]: Can you make a for-loop here as well?

File: tests/ignite/metrics/regression/test_median_absolute_error.py [hunk 3]
```diff
@@ -34,38 +34,36 @@ def test_wrong_input_shapes():
...
 
     m.reset()
     m.update((y_pred, y))
 
-    assert np_median_absolute_error == pytest.approx(m.compute())
+    assert expected_median_absolute_error == pytest.approx(m.compute())
 
 
-def test_median_absolute_error_2():
+def test_median_absolute_error_2(available_device):
```
Comment (vfdev-5) [inline tests/ignite/metrics/regression/test_median_absolute_error.py:57]: Let's remove the line below: `np.random.seed(1)`

Comment (vfdev-5) [reply tests/ignite/metrics/regression/test_median_absolute_error.py:57]: Check this one

File: tests/ignite/metrics/regression/test_mean_normalized_bias.py [hunk 4]
```diff
@@ -38,77 +38,58 @@ def test_wrong_input_shapes():
...
-        test_cases = get_test_cases()
-        for y_pred, y, batch_size in test_cases:
-            _test(y_pred, y, batch_size)
+def test_mean_error(available_device):
+    preds = [torch.randn(4) for _ in range(4)]
+    target = torch.randn(4)
+
+    if available_device == "mps":
+        preds = [p.to(dtype=torch.float32) for p in preds]
+        target = target.to(dtype=torch.float32)
```
Comment (vfdev-5) [inline tests/ignite/metrics/regression/test_mean_normalized_bias.py:47]:
```suggestion
```

**Review by vfdev-5 — APPROVED** [review]
